spec/tictactoe_spec.rb

```diff
@@ -8,73 +8,57 @@ class FakeIO
   def initialize(stdin: '', stdout: [])
     @stdin = stdin
     @stdout = stdout
+    @accepted_input = [*1..9].map!(&:to_s)
   end
 
-  def puts(string)
+  def output_to_user(string)
     @stdout << string
   end
 
-  def gets
-    string_to_return = @stdin
+  def prompt_user_for_position
+    string_to_return = @accepted_input.include?(@stdin) ? @stdin.to_i - 1: nil
```

Comment:
A rule of thumb I like to follow is that by test double shouldn't contain control flow. `If` statements make me wonder if this test object needs its own tests!

Reply:
Yeah, this fake looks just about as complicated as the real thing, and seems to behave about the same too. The thing to realize with a fake is all you really have to do is stick to the same interface as the real thing (i.e. public method names, inputs, and outputs), but within those constraints you can often keep things quite simple. One helpful thing to remember is because you're injecting it, by definition, the constructor isn't used by the class under test, so you're free to define one or more constructor methods for your Fake that set it up to behave however you'd like.